parser/metadecoders/decoder.go

```diff
@@ -122,7 +128,12 @@ func (d Decoder) Unmarshal(data []byte, f Format) (any, error) {
 	if len(data) == 0 {
 		switch f {
 		case CSV:
-			return make([][]string, 0), nil
+			if d.ToMap {
+				return make(map[string]any), nil
+			} else {
+				return make([][]string, 0), nil
+			}
```

Comment:
The `else` is unnecessary since the `if` will return early.

```go
if d.ToMap {
	return make(map[string]any), nil
}
return make([][]string, 0), nil
```